### PR TITLE
Enforce dhm_min_bitlen exactly

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
In a TLS client, enforce the Diffie-Hellman minimum parameter size set with `mbedtls_ssl_conf_dhm_min_bitlen()` precisely. Before, the minimum size was rounded down to the nearest multiple of 8.

This is a very minor bug, but I found this incorrect line of code distracting.

Backports:
* 2.x: https://github.com/ARMmbed/mbedtls/pull/4430
* 2.16: optional. It's a very minor bug, but the risk is low and having LTS branches aligned makes future backports easier. https://github.com/ARMmbed/mbedtls/pull/4325

### CODE REVIEW

File: tests/data_files/Makefile [hunk 1]
```diff
@@ -1131,6 +1131,16 @@ cert_md5.crt: cert_md5.csr
 	$(MBEDTLS_CERT_WRITE) request_file=$< serial=6 issuer_crt=$(test_ca_crt) issuer_key=$(test_ca_key_file_rsa) issuer_pwd=$(test_ca_pwd_rsa) not_before=20000101121212 not_after=20300101121212 md=MD5 version=3 output_file=$@
 all_final += cert_md5.crt
 
+################################################################
+#### Diffie-Hellman parameters
+################################################################
+
+dh.998.pem:
+	$(OPENSSL) dhparam -out $@ -text 998
```
Comment (mstarzyk-mobica) [inline tests/data_files/Makefile:1139]: I dont' quite understand this. Isn't that generating the dh.998.pem file, the one that is also added in this PR?

Comment (gilles-peskine-arm) [reply tests/data_files/Makefile:1139]: Yes, that's by design. See the comment at the top of the makefile.

Comment (mstarzyk-mobica) [reply tests/data_files/Makefile:1139]: ooooh. Ok, that makes sense now, thank you :)

**Review by mstarzyk-mobica — APPROVED** [review]
I'm not sure about the 999 and 998 files relation to the mentioned (in the commit message) byte-aligned boundary.
Code looks ok.

**Review by mpg — APPROVED** [review]
Looks good to me, thanks for fixing. I vote we backport as this shouldn't be too much effort.

**Comment by mstarzyk-mobica** [comment]
CI complains that branch is missing:
`fatal: Remote branch dev/gilles-peskine-arm/dhm_min_bitlen-bits not found in upstream origin`
Is it just a CI hickup?

**Comment by gilles-peskine-arm** [comment]
“Travis CI - Branch” should not have run at all. It ran because I accidentally pushed some branches to the repository. Please ignore it.

**Pull request merged**